xline/src/storage/execute_error.rs

```diff
@@ -0,0 +1,134 @@
+use thiserror::Error;
+
+/// Error met when executing commands
+// #[allow(clippy::module_name_repetitions)] // this-error generate code false-positive
+// #[non_exhaustive]
+#[derive(Error, Debug, Clone)]
+pub(crate) enum ExecuteError {
+    /// Kv error
+    #[error("kv error: {0}")]
+    KvError(String),
+    /// Lease error
+    #[error("lease error: {0}")]
+    LeaseError(String),
+    /// Auth error
+    #[error("auth error: {0}")]
+    AuthError(String),
+    /// Db error
+    #[error("db error: {0}")]
+    DbError(String),
+    /// Permission denied
+    #[error("permission denied")]
+    PermissionDenied,
+}
+
+impl ExecuteError {
+    /// Create a new `ExecuteError::DbError`
+    pub(crate) fn db_error(message: impl Into<String>) -> Self {
+        Self::DbError(message.into())
+    }
+
+    /// Key not found
+    pub(crate) fn key_not_found() -> Self {
+        Self::KvError("key not found".to_owned())
+    }
+
+    /// Lease not found
+    pub(crate) fn lease_not_found(lease_id: i64) -> Self {
+        Self::LeaseError(format!("lease {} not found", lease_id))
+    }
+
+    /// Lease is expired
+    pub(crate) fn lease_expired(lease_id: i64) -> Self {
+        Self::LeaseError(format!("lease {} is expired", lease_id))
+    }
+
+    /// Lease ttl is too large
+    pub(crate) fn lease_ttl_too_large(ttl: i64) -> Self {
+        Self::LeaseError(format!("lease ttl is too large: {}", ttl))
+    }
+
+    /// Lease already exists
+    pub(crate) fn lease_already_exists(lease_id: i64) -> Self {
+        Self::LeaseError(format!("lease {} already exists", lease_id))
+    }
+
+    /// Auth is not enabled
+    pub(crate) fn auth_not_enabled() -> Self {
+        Self::AuthError("auth is not enabled".to_owned())
+    }
+
+    /// Auth Failed
```

Comment:
Auth failed. Plz keep the comment style consistent.

Reply:
fixed